### PR TITLE
gss/sss (solid harmonic ordering) runtime switchable

### PR DESCRIPTION
**Pull request by loriab**
(in order of decreasing importance)

- [x] add runtime switchable solid harmonic ordering
  - [x] in initialize.h, add setter and accessor so that one can do the below (as psi4 does: https://github.com/psi4/psi4/blob/master/psi4/src/core.cc#L1038) . If SH ordering is never set, default is standard ordering.
  ```
    libint2::initialize();
    libint2::set_solid_harmonics_ordering(libint2::SHGShellOrdering_Gaussian);
  ``` 
  - [x] in shgshell_ordering, inlines for both orderings are now always defined. `INT_SOLIDHARMINDEX(l, m)` with 2 args has been deprecated (throws a warning) but still returns the value as the library was configured at build-time. a new `INT_SOLIDHARMINDEX(sho, l, m)` with 3 args returns the value according to the SH ordering requested in the new 1st arg.
  - [x] in `solidharmonics.h`, generating the indices switches from ifdefs to logic.
* [added 3 Sept] see https://github.com/evaleev/libint/pull/271/commits/05ff2f12e35dbce950cc1faa586e60367385a05e for testing of the runtime switchable SHO functionality.
- [x] the only non-runtime-switchable use of solid harmonics is for generating the multipole ints code in build_libint.cc . we discussed possibly removing that option and settling on STANDARD. but for now to be minimally disruptive, it's still configurable. 
- [x] more migration from `find_package(PythonInterp)` to ` find_package(Python COMPONENTS Interpreter Development)`. Specify directly with `Python_EXECUTABLE`. Hint behavior is complex, as described in https://cmake.org/cmake/help/latest/module/FindPython.html .
- [x] to use the new Python detection, it's recommended to require cmake 3.15+ (https://pybind11.readthedocs.io/en/stable/compiling.html?highlight=pybind11_nopython#findpython-mode). since the final goal is 3.16, I went ahead and bumped it to that.
- [x] add in configuration.h/cc files . at the moment, all this does is add a function `configuration_accessor()` that returns `(nyi)`. this is in preparation for later accessing build configration at runtime.
- [x] expand GHA testing. the first stage `build_repo` (was `build`) remains largely the same except that the `Release` lanes upload a GHA artifact of the exported tarball. in the second stage, the _Linux_ artifact gets downloaded into a fresh environment, and the library is built with gcc and icpx on Linux and clang on Mac. These all use a conda environment for uniformity across platforms. (the first stage is testing homebrew and apt-get.) only the `hf++` test is run. python bindings are built and tested.
- [x] bump GHA Ununtu gcc from 9 to 10

* Note that this snapshot has not been tested against Psi4. At the moment, it's the full #259 implementation that I know to be working.

* Now rebased to master. Please squash PR when merging.

### CODE REVIEW

File: python/src/libint2/libint2.cc [hunk 1]
```diff
@@ -137,6 +137,12 @@ std::vector<double> coeffs_normalized(const Shell &s) {
 
 PYBIND11_MODULE(libint2, m) {
 
+  py::enum_<SHGShellOrdering>(m, "SHGShellOrdering")
+    .value("SHGShellOrdering_Standard", libint2::SHGShellOrdering_Standard)
+    .value("SHGShellOrdering_Gaussian", libint2::SHGShellOrdering_Gaussian)
+    .value("SHGShellOrdering_MOLDEN", libint2::SHGShellOrdering_Gaussian)
```
Comment (loriab) [inline python/src/libint2/libint2.cc:143]: On further investigation, I don't think this is a typo. The Molden: Gaussian mirrors https://github.com/evaleev/libint/blob/master/include/libint2/shgshell_ordering.h#L33 , and the defines in that file do seem to match, too. 

I'm guessing it was a convenience to have Molden always defined even when the main SH value toggled.

**Review by evaleev — APPROVED** [review]

**Comment by evaleev** [comment]
@loriab looks good, I only tweaked this slightly (spherical multipoles operator components use standard order always ... this makes sense by analogy with the cartesian multipole operator components not depending on the Cartesian shell order)

**Comment by evaleev** [comment]
@loriab which is the next PR in the laundry list?

**Comment by loriab** [comment]
Great, thanks! @270 is next and presently includes the commits of this one, so lmk if you'd like me to rebase.

**Comment by evaleev** [comment]
Sure that would be helpful. Also if you want to squash related commits (try
xxx) feel free ... I am not prmedabtic/paranoid about the history ...

**Comment by loriab** [comment]
> Sure that would be helpful. Also if you want to squash related commits (try
xxx) feel free ... I am not prmedabtic/paranoid about the history ...

Sure, I'll fix up 270 after this one is merged (GHA just completed). All the "try xxx" are long gone -- it's actually reviewable commit-by-commit if you prefer.

**Comment by evaleev** [comment]
@loriab heads up: re-enabled `FOR_SOLIDHARM` to be able to write generic code that can work with any Libint2 library

**Pull request merged**